src/Converter.cpp

```diff
@@ -237,357 +302,208 @@ void Converter::printBegin()
-    case 2:
-        numElements = *reinterpret_cast<uint16_t*>(data + offset);
-        break;
-    case 4:
-        numElements = *reinterpret_cast<uint32_t*>(data + offset);
-        break;
-    case 8:
-        numElements = *reinterpret_cast<uint64_t*>(data + offset);
-        break;
-    default:
-        throw std::runtime_error(
-            "Unknown size type size, should be one of 1, 2, 4, 8");
+    size_t numElements = type.getElementCount(v.getData());
+    
+    if (debug) {        
+        printBegin(); std::cout << __FUNCTION__ << "Container at "<< v.getData() << std::endl;    
+        printBegin(); std::cout << "numElements: " <<numElements << std::endl;
+        printBegin(); std::cout << type.kind() << "[" << numElements << "]" << std::endl;
     }
-    offset += size;
-
-    if(numElements > containerLimit)
-    {
-        // HACK Sometimes we are starting at the wrong offset, we have to
-        //      ignore some padding bytes... Don't try this at home!
-        size_t paddingOffset = 0;
-        const size_t MAXIMUM_ALLOWED_PADDING = 7;
-        for(paddingOffset = 0; paddingOffset <= MAXIMUM_ALLOWED_PADDING;
-            paddingOffset++)
-        {
-            if(*(data + offset - size + paddingOffset) != 0)
-                break;
-        }
-        size_t paddedNumElements = 0;
-        switch(size)  // TODO refactor
-        {
-        case 1:
-            paddedNumElements = *reinterpret_cast<uint8_t*>(
-                data + offset - size + paddingOffset);
-            break;
-        case 2:
-            paddedNumElements = *reinterpret_cast<uint16_t*>(
-                data + offset - size + paddingOffset);
-            break;
-        case 4:
-            paddedNumElements = *reinterpret_cast<uint32_t*>(
-                data + offset - size + paddingOffset);
-            break;
-        case 8:
-            paddedNumElements = *reinterpret_cast<uint64_t*>(
-                data + offset - size + paddingOffset);
-            break;
-        default:
-            throw std::runtime_error(
-                "Unknown size type size, should be one of 1, 2, 4, 8");
-        }
-        bool padded = paddedNumElements < containerLimit;
-        std::cerr << "[pocolog2msgpack] WARNING: container limit exceeded, ";
-        if(padded)
-        {
-            numElements = paddedNumElements;
-            offset += paddingOffset;
-            std::cerr << "ignoring " << paddingOffset << " padding bytes"
-                << std::endl;
-        }
-        else
-        {
-            numElements = containerLimit;
-            std::cerr << "truncating " << type.kind() << "! (" << numElements
-                << " > " << containerLimit << ")" << std::endl;
-        }
-    }
-
-    if(!part && verbose >= 3 + depth)
-    {
-        printBegin();
-        std::cout << type.kind() << "[" << numElements << "]" << std::endl;
+    
+    if(numElements > containerLimit) {
+        throw std::runtime_error("too many elements");
```

Comment:
I think a warning message would be better than an error. Maybe this is intended behavior to cut the converted log.